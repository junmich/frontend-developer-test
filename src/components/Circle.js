import React from 'react';

const Circle = (props) => {
    return <div className={props.circleClass}>{props.name}</div>;
}

Circle.defaultProps = {
    name: '',
    circleClass: 'circle deg-200-10'
}

export default Circle;