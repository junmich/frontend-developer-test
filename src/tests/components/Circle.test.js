import React from 'react';
import { shallow } from 'enzyme';
import Circle from '../../components/Circle';

test('should render circle correctly', () => {
    const wrapper = shallow(<Circle />);
    expect(wrapper).toMatchSnapshot();
});

test('should render circle and assign circle class correctly', () => {
    const wrapper = shallow(<Circle circleClass="testClass" />);
    expect(wrapper.find('.testClass').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
});