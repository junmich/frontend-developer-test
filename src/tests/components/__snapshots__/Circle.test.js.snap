// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render circle and assign circle class correctly 1`] = `
<div
  className="testClass"
/>
`;

exports[`should render circle correctly 1`] = `
<div
  className="circle deg-200-10"
/>
`;
